packages/datetime/src/dateRangeInput.tsx

```diff
@@ -128,9 +165,19 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         );
     }
 
+    public componentWillReceiveProps(nextProps: IDateRangeInputProps) {
+        if (nextProps.value !== this.props.value) {
+            const [selectedStart, selectedEnd] = this.getInitialRange(nextProps);
+            this.setState({ selectedStart, selectedEnd });
+        }
+        super.componentWillReceiveProps(nextProps);
```

Comment:
That's what my instinct was, but `DateInput` did it this way, so I mimicked. From `dateInput.tsx`:

```
public componentWillReceiveProps(nextProps: IDateInputProps) {
    if (nextProps.value !== this.props.value) {
        this.setState({ value: fromDateToMoment(nextProps.value) });
    }

    super.componentWillReceiveProps(nextProps);
}
```

Will change though.